src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -691,7 +692,16 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
                         Debug.Assert(stats.GC.m_stats.suspendTimeRelativeMSec != -1);
                         if (isEphemeralGCAtBGCStart || _gc.Reason == GCReason.PMFullGC)
                         {
-                            _gc.PauseStartRelativeMSec = data.TimeStampRelativeMSec;
+                            if (isEphemeralGCAtBGCStart)
+                            {
+                                // in that case, we don't want to lose the suspension time elapsed since the BGC started
+                                _gc.PauseStartRelativeMSec = stats.GC.m_stats.suspendTimeRelativeMSec;
```

Comment:
instead of counting it we should initialize the BGC pause a few lines below
```
                        _gc.StartRelativeMSec = data.TimeStampRelativeMSec;
                        stats.GC.GCs.Add(_gc);
                        if (_gc.Type == GCType.BackgroundGC)
                        {
                            stats.GC.m_stats.currentBGC = _gc;
                            // For BGC, we need to add the suspension time so far to its pause so we don't miss including it.
                            // If there's an ephemeral GC happening before the BGC starts, AddConcurrentPauseTime will not
                            // add this suspension time to GC pause as that GC would be seen the ephemeral GC, not the BGC.
                            _gc.PauseDurationMSec = _gc.SuspendDurationMSec;
                            _gc.ProcessCpuAtLastGC = stats.GC.m_stats.ProcessCpuAtLastGC;
                        }
```

Reply:
You're right: sounds more consistent to me